charts/templates/statefulset.yaml

```diff
@@ -31,83 +31,29 @@ spec:
+              value: "true"
+            - name: REPLICA_COUNT
+              value: {{ .Values.replicaCount  | quote }}
           envFrom:
             - configMapRef:
                 name: {{ include "marklogic.fullname" . }}
-          command:
-            - bash
-            - '-c'
-            - |
-              echo '### Begin ML Container Config ###'
-
-              # Set Hostname to the value of hostname command to /etc/marklogic.conf
-              HOST_FQDN="$(hostname).$ML_FQDN_SUFFIX"
-              echo "export MARKLOGIC_HOSTNAME=\"$HOST_FQDN\"" | sudo tee /etc/marklogic.conf
-
-              cd ~
-
-              AUTH_CURL="curl --anyauth --user $ML_ADMIN_USER:$ML_ADMIN_PASS -m 20 -s "
-
-              # Start MarkLogic
-              sudo service MarkLogic start
-              sleep 5s
-
-              # Initialize and Setup Admin User
-              echo '### Initialize and Setup Admin User ###'
-              ML_STATUS_CODE=`curl -s -o /dev/null --write-out %{response_code}  http://localhost:8001/admin/v1/init`
-              if [ "$ML_STATUS_CODE" == "401" ]; then
-                echo "Server is already configured." 
-              else
-                curl -X POST -d "" http://$HOSTNAME:8001/admin/v1/init
-                sleep 5s
-
-                curl -X POST -H "Content-type: application/x-www-form-urlencoded" \
-                    --data "admin-username=$ML_ADMIN_USER" --data "admin-password=$ML_ADMIN_PASS" \
-                    --data "realm=public" \
-                    http://$HOSTNAME:8001/admin/v1/instance-admin
-                sleep 10s
-
-                # for EA1 release, disable turning on XDQP by default.
-                # Turn on XDQP encryption via CMA
-                # $AUTH_CURL -X POST -H 'Content-Type: application/json' \
-                #     -d '{"config":[{"op":"update","group":[{"group-name":"Default","xdqp-ssl-enabled":true}]}]}' \
-                #     http://$HOSTNAME:8002/manage/v3/
-
-                # Join Cluster
-                if [ "$HOSTNAME" != "$ML_BOOTSTRAP_HOST" ]; then
-                  echo "### joining cluster ###"
-                  joiner=$HOSTNAME
-                  cluster="$ML_BOOTSTRAP_HOST"
-                  $AUTH_CURL -o host.xml -X GET -H "Accept: application/xml" http://${joiner}:8001/admin/v1/server-config
-                  $AUTH_CURL -X POST -d "group=Default" --data-urlencode "server-config@./host.xml" -H "Content-type: application/x-www-form-urlencoded" -o cluster.zip http://${cluster}:8001/admin/v1/cluster-config
-
-                  sleep 10s
-
-                  $AUTH_CURL -X POST -H "Content-type: application/zip" --data-binary @./cluster.zip http://${joiner}:8001/admin/v1/cluster-config 
-                  sleep 5s
-
-                  rm -f host.xml
-                  rm -f cluster.zip
-                fi
-
-              fi
-
-              # mark the pod ready for readiness probe
-              sudo touch /var/opt/MarkLogic/ready
```

Comment:
You will need to add this logic to the docker script so the current readiness probe works

Reply:
Yes, I did, see PR 86 in the docker repo.

Reply:
I think it's better to have the logic remain in the Kubernetes side for the following reasons:
1. This logic only works for Kubernetes. For standalone docker image this ready file doesn't make sense.
2. This is a temporary solution. If we want to change this behavior we should only change it in Kubernetes statefulset.yaml file, instead of have external dependency on docker image.

Reply:
This file is a way to track state of when the MarkLogic instance has been initialized and joined a cluster (if told to). That is not specific to k8s. Right now, this is the only criteria for "ready" on the k8s side. If need to revisit that, we can add logic to the readiness probe implementation where there state of this flag can be part of that logic.

Thinking about this a bit more, it would be good to be able to track the state of each step of the startup script so we know where it is in the process. The readiness probe can just look for the container to reach the right state to be considered "ready". Once again, that is not necessarily specific to k8s.

The XDQB encryption setting is a MarkLogic group-level setting so we should discuss the group-level configuration parameters we need to support as this will be one of them. When a host joins a cluster, the group needs to be specified. It current uses "Default" as that is the default. Both of these things should be in the container "API", not just in k8s.